### PR TITLE
Add Modal, and HxDiv elements & Alert event fixes

### PR DESCRIPTION
**Pull request by nicko-winner**
- Create useEventListener hook for more terse hook creation
 - Favoring shorter smaller wrappers with more ...rest passing, relying on prop types for prop documentation, Webstorm seems to handle this less explicit approach well.
 - Add actions to Alert, and update update the event names we are listening for to be correct
 - Move constants to a single file.

### CODE REVIEW

File: src/hooks/useEventListener.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+function useEventListener(eventHandlers = {}, ref) {
+  const theRef = ref || useRef(null);
+  useEffect(() => {
+    Object.entries(eventHandlers).forEach(([handlerName, eventHandler], key) => {
+      theRef.current.addEventListener(handlerNameToEvent(handlerName), eventHandler);
+    });
+    return () => {
+      Object.entries(eventHandlers).forEach(([handlerName, eventHandler], key) => {
+        theRef.current.addEventListener(handlerNameToEvent(handlerName), eventHandler);
```
Comment (michaelmang) [inline src/hooks/useEventListener.js:19]: Did you mean `removeEventListener`?

File: src/Modal/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import useEventListener from '../hooks/useEventListener';
+import { SIZES } from '../constants';
+
+const Modal = ({ onOpen, onClose, className, open, size, children, ...rest }) => {
+  const hxRef = useEventListener({ onOpen, onClose });
+  return (
+    <hx-modal
+      class={classNames(className, SIZES[size])}
+      ref={hxRef}
+      open={open ? true : null}
```
Comment (michaelmang) [inline src/Modal/index.js:13]: I was wondering if there'd be a good util function to make to communicate why this is being done.

Some ideas:

`noFalse`
`nullifyFalse`
`treatFalseAsNull`
`supportFalse`

If we can't think of a good name, it's fine to leave as is

File: src/hooks/useEventListener.js (hunk 1 above)
Comment (nicko-winner) [reply src/hooks/useEventListener.js:19]: good catch.

File: src/Modal/index.js (hunk 2 above)
Comment (nicko-winner) [reply src/Modal/index.js:13]: `wcBool()`  - short for web component boolean ? 
`wcBooleanAttribute()`  more descriptive, but kind of long.

Comment (michaelmang) [reply src/Modal/index.js:13]: I like it, a middleground for your second option would be `wcBoolAttr`

**Pull request merged**